packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/react/basic-sign-in.md

```diff
@@ -0,0 +1,202 @@
+      setUserInfo(null);
+    } else {
+      oktaAuth.getUser().then((info) => {
+        setUserInfo(info);
+      }).catch((err) => {
+        console.error(err);
+      });
+    }
+  }, [authState, oktaAuth]); // Update if authState changes
+
+  if (!userInfo) {
+    return (
+      <div>
+        <p>Fetching user info ...</p>
+      </div>
+    );
+  }
+
+  return (
+    <div>
+      <div>
+        <p id="welcome">
+          Welcome, &nbsp;{userInfo.name}!
+        </p>
+        <p>You have successfully authenticated against your Okta org, and have been redirected back to this application.</p>
+      </div>
+    </div>
+  );
+};
+
+export default Protected;
+```
+
+#### Login route
+
+This route hosts the Sign-In Widget and redirects the user to the default home page if the user is already signed in.
+
+For example, create a `src/Login.jsx` file with the Login route component:
+
+```js
+import React from 'react';
+import { Redirect } from 'react-router-dom';
+import OktaSignInWidget from './OktaSignInWidget';
+import { useOktaAuth } from '@okta/okta-react';
+
+const Login = ({ config }) => {
+  const { oktaAuth, authState } = useOktaAuth();
+  const onSuccess = (tokens) => {
+    oktaAuth.handleLoginRedirect(tokens);
+  };
+
+  const onError = (err) => {
+    console.log('Sign in error:', err);
+  };
+
+  if (!authState) {
+    return <div>Loading ... </div>;
+  }
+
+  return authState.isAuthenticated ?
+    <Redirect to={{ pathname: '/' }}/> :
+    <OktaSignInWidget config={config} onSuccess={onSuccess} onError={onError}/>;
+};
+export default Login;
+```
+
+#### Callback route
+
+The [Okta React SDK](https://github.com/okta/okta-react) provides the [LoginCallback](https://github.com/okta/okta-react#logincallback) component for the callback route. It handles token parsing, token storage, and redirects the user to the `/` path. If a `SecureRoute` triggered the redirect, then the callback is directed to the secured route. See how the callback route component is called from the route definition file (`src/App.jsx`) in the [Connect the routes](#connect-the-routes) section.
+
+### Connect the routes
+
+Update `src/App.jsx` to include your project components and routes. The `Security` component controls the authentication flows, so it requires your OpenID Connect configuration. By default, `@okta/okta-react` redirects to Okta's sign-in page when the user isn't authenticated. In this example, `onAuthRequired` is overridden to redirect to the custom sign-in route instead.
+
+> **Note:** This example uses the `react-router-dom` version 5.x module for the `useHistory` and `Route` objects.
+
+```js
+import React from 'react';
+import { Route, useHistory } from 'react-router-dom';
+import { Security, SecureRoute, LoginCallback } from '@okta/okta-react';
+import { OktaAuth, toRelativeUrl } from '@okta/okta-auth-js';
+import Home from './Home';
+import Login from './Login';
+import Protected from './Protected';
+import Profile from './Profile';
```

Comment:
Remove this import or there's an error:

```
Compiled with problems:X

ERROR in ./src/App.js 13:0-32

Module not found: Error: Can't resolve './Profile' in '/Users/mraible/Downloads/okta-app/src'
```

Reply:
opps.... sorry, was testing a few things. I'll remove that line.